### PR TITLE
static analysis of networks, renamed graphs to networks, improvements on the excluding clause

### PR DESCRIPTION
**Pull request by mcodescu**
Should solve #1331, also #1467 

A test file is here: https://ontohub.org/dol-examples/networks.dol, another one here: https://ontohub.org/sandbox/networks.dol

For excluding paths, I will have to merge Christian's branch.

### CODE REVIEW

File: Static/AnalysisLibrary.hs [hunk 1]
```diff
@@ -482,6 +483,21 @@ anaLibItem lg opts topLns currLn libenv dg eo itm =
...
+        diag = makeDiagram dg cNodes cEdges
+        genv = globalEnv dg
+    if Map.member nn genv 
+       then
+        liftR $ plain_error (itm, dg, libenv, lg, eo)
+               (alreadyDefined nnstr) pos
+       else
+        return (itm, dg{globalEnv = Map.insert nn (NetworkEntry diag) genv},
+                libenv, lg, eo)
+    --error "static analysis of network not yet implemented"
```
Comment (tillmo) [inline Static/AnalysisLibrary.hs]: this comment should go

File: Static/AnalysisStructured.hs [hunk 2]
```diff
@@ -631,11 +646,20 @@ anaSpecAux conser addSyms lg libEnv ln dg nsig name opts eo sp rg = case sp of
...
                          [lEdge s1 b i1, lEdge s1 t1 sig1,
                           lEdge s2 b i2, lEdge s2 t2 sig2] ++ cE)
+               NetworkEntry diag -> let
+                    dnodes = nodes diag
+                    ledges = labEdges diag
+                    dgedges = map (\(x,y, (_, m)) -> lEdge x y m) ledges
+                   in (dnodes, dgedges)
                _ -> error $ show cItem
-                    ++ "is not an ontology, a view or an alignment"
+                    ++ " is not an ontology, a view, a network or an alignment"
```
Comment (tillmo) [inline Static/AnalysisStructured.hs]: ontology -> OMS (ontology, model, or specification)

File: Static/AnalysisStructured.hs [hunk 3]
```diff
@@ -596,7 +597,17 @@ anaSpecAux conser addSyms lg libEnv ln dg nsig name opts eo sp rg = case sp of
                    (replaceAnnoted sp2' asp2)
                    pos, nsig3, udg3)
   Combination (Network cItems eItems _) pos -> adjustPos pos $ do
-    let getNodes (cN, cE) cItem = let
+    let (cNodes', cEdges') = networkDiagram dg cItems eItems
+    (ns, dg') <- insertColimitInGraph libEnv dg cNodes' cEdges' name
+    return (sp, ns, dg')
+  _ -> fail $ "AnalysisStructured: " ++ show (prettyLG lg sp)
+
+networkDiagram :: DGraph
```
Comment (tillmo) [inline Static/AnalysisStructured.hs]: at least such top-level functions should be documented with their purpose

File: Static/AnalysisStructured.hs [hunk 4]
```diff
@@ -596,7 +597,17 @@ anaSpecAux conser addSyms lg libEnv ln dg nsig name opts eo sp rg = case sp of
...
+                        -> [LABELED_ONTO_OR_INTPR_REF]
+                        -> [IRI]
+                        -> ([Node], [(Node, Node, DGLinkLab)])
+networkDiagram dg cItems eItems = let 
+        getNodes remove (cN, cE) cItem = let
```
Comment (tillmo) [inline Static/AnalysisStructured.hs]: ...however, even for such local functions, documentation with their purpose would be helpful.

**Comment by tillmo** [comment]
see my comments on the code

**Comment by mcodescu** [comment]
Done.

**Comment by tillmo** [comment]
:+1:

**Pull request merged**